python/pynq/mmio.py

```diff
@@ -96,16 +97,18 @@ def __init__(self, base_addr, length = 4, debug = False):
         self.length = length
         
         self.debug = debug
-        self._debug('MMIO(address, size) = ({0}, {1} bytes).'.\
-                    format(hex(self.base_addr), hex(self.length)))
+        self._debug('MMIO(address, size) = ({0:x}, {1:x} bytes).',
+                    self.base_addr,self.length)
                 
         # Open file and mmap
-        f = os.open(general_const.MMIO_FILE_NAME, os.O_RDWR | os.O_SYNC)
-        self.mem = mmap.mmap(f, (self.length + self.virt_offset), 
+        self.f = os.open(general_const.MMIO_FILE_NAME, os.O_RDWR | os.O_SYNC)
+        self.mem = mmap.mmap(self.f, (self.length + self.virt_offset),
```

Comment:
We ought to close the file when the MMIO object gets cleans up, hence why I turned it into an attribute. I forgot to check in the close method. Thoughts on an explicit `close` vs. using `__del__` welcome. I vote for using `__del__` as it isn't essential we always clean up, it would just be preferable to not have 10s or 100s of file descriptors pointing at `/dev/mem` while playing around with driver development.